OpenTabletDriver/Reflection/PluginManager.cs

```
@@ -0,0 +1,97 @@
+                where IsPluginType(type)
+                where IsPlatformSupported(type)
+                select type;
+
+            pluginTypes = new ConcurrentBag<TypeInfo>(internalTypes);
+        }
+
+        public IReadOnlyCollection<TypeInfo> PluginTypes => pluginTypes;
+        protected readonly ConcurrentBag<TypeInfo> pluginTypes;
+
+        protected readonly static IEnumerable<Type> libTypes =
+            from type in Assembly.GetAssembly(typeof(IDriver)).GetExportedTypes()
+                where type.IsAbstract || type.IsInterface
+                select type;
+
+        public virtual PluginReference GetPluginReference(string path) => new PluginReference(this, path);
+        public virtual PluginReference GetPluginReference(Type type) => GetPluginReference(type.FullName);
+        public virtual PluginReference GetPluginReference(object obj) => GetPluginReference(obj.GetType());
+
+        public virtual T ConstructObject<T>(string name, object[] args = null) where T : class
+        {
+            args ??= new object[0];
+            if (!string.IsNullOrWhiteSpace(name))
+            {
+                try
+                {
+                    var type = PluginTypes.FirstOrDefault(t => t.FullName == name);
+                    var matchingConstructors = from ctor in type?.GetConstructors()
+                        let parameters = ctor.GetParameters()
+                        where parameters.Length == args.Length
+                        where IsValidParameterFor(args, parameters)
+                        select ctor;
+
+                    var constructor = matchingConstructors.FirstOrDefault();
+                    return (T)constructor?.Invoke(args) ?? null;
+                }
+                catch
+                {
+                    Log.Write("Plugin", $"Unable to construct object '{name}'", LogLevel.Error);
+                }
+            }
+            return null;
```

Probably would be better to throw an exception instead of returning null, could reduce some pains.

---

Will change this later in a different PR where needed. Too much depends on this returning null, so we just will leave it be for now.